### PR TITLE
Add shadow on TagView

### PR DESCRIPTION
**Pull request by kitasuke**
I have to use a new UIView to draw shadow, because corner radius and shadow don't work well on same layer.

### CODE REVIEW

File: TagListView/TagListView.swift [hunk 1]
```diff
@@ -156,7 +174,7 @@ public class TagListView: UIView {
         var currentRowView: UIView!
         var currentRowTagCount = 0
         var currentRowWidth: CGFloat = 0
-        for tagView in tagViews {
+        for (index,tagView) in tagViews.enumerate() {
```
Comment (xhacker) [inline TagListView/TagListView.swift]: Can you add a space after the comma?

File: TagListView/TagListView.swift [hunk 2]
```diff
@@ -171,8 +189,17 @@ public class TagListView: UIView {
                 addSubview(currentRowView)
             }
             
-            tagView.frame.origin = CGPoint(x: currentRowWidth, y: 0)
-            currentRowView.addSubview(tagView)
+            let tagBackgroundView = tagBackgroundViews[index]
+            tagBackgroundView.frame.origin = CGPoint(x: currentRowWidth, y: 0)
+            tagBackgroundView.frame.size = tagView.bounds.size
+            tagBackgroundView.layer.shadowColor = shadowColor.CGColor
+            tagBackgroundView.layer.shadowPath = UIBezierPath(roundedRect: tagBackgroundView.bounds, cornerRadius: shadowRadius).CGPath
```
Comment (xhacker) [inline TagListView/TagListView.swift]: We should use `cornerRadius` instead of `shadowRadius` here, right?

**Comment by xhacker** [comment]
Looks good, just some minor comments. I’m curious why do you need a `tagBackgroundView`?

**Comment by kitasuke** [comment]
Shadow didn't work without a new UIView if cornerRadius was set.
Please see more details below.
http://stackoverflow.com/questions/24644802/swift-problems-with-corner-radius-and-drop-shadow

File: TagListView/TagListView.swift (hunk 2 above)
Comment (kitasuke) [inline TagListView/TagListView.swift]: No.
According to the official document, `shadowRadius` means `The blur radius used to create the shadow. Defaults to 3. Animatable.` So it's different concept with `cornerRadius`.
You will see if you change to cornerRadius and run this project!

Comment (xhacker) [inline TagListView/TagListView.swift]: I mean in
```swift
tagBackgroundView.layer.shadowPath = UIBezierPath(roundedRect: tagBackgroundView.bounds, cornerRadius: shadowRadius).CGPath
```
If it’s different, the shadow would have different corner radius.

screen shot 2015-12-22 at 16 06 52

Comment (kitasuke) [inline TagListView/TagListView.swift]: Oh you are right.
I've just fixed it.

**Comment by xhacker** [comment]
Can you squash the last two commits into the previous ones? Then I’ll merge. :)

**Comment by kitasuke** [comment]
Let me confirm what I should do. Do you want me to revert `0b57e8d` and `99c2e52`, then make them one commit?

**Comment by xhacker** [comment]
I mean to squash e64741d37b14850708b45fb4ca374d03394e2ddd, 99c2e52e1bea6de0150d32167563d5b90e4afd31, and 0b57e8d91ddf0dbb1cea4cf382ea6f2c11f20d93 into one commit. You can use `git rebase -i`.

**Comment by kitasuke** [comment]
Done!

**Comment by xhacker** [comment]
Awesome, thank you!

**Pull request merged**